src/functions.jl

```diff
@@ -173,21 +463,64 @@ function VectorQuadraticTerm(
+end
+
+term_pair(t::VectorQuadraticTerm) = term_indices(t) => coefficient(t)
+
 """
-    VectorQuadraticFunction{T}(quadratic_terms, affine_terms, constants)
+    VectorQuadraticFunction{T}(
+        quadratic_terms::Vector{VectorQuadraticTerm{T}},
+        affine_terms::Vector{VectorAffineTerm{T}},
+        constants::Vector{T},
+    ) where {T}
 
 The vector-valued quadratic function with i`th` component ("output index")
-defined as ``\\frac{1}{2}x^TQ_ix + a_i^T x + b_i``, where:
-* ``a_i`` is a sparse vector specified by the `VectorAffineTerm`s with
-  `output_index == i`.
-* ``b_i`` is a scalar specified by `constants[i]`
-* ``Q_i`` is a symmetric matrix specified by the `VectorQuadraticTerm` with
-  `output_index == i`.
-
-Duplicate indices in ``a_i`` or ``Q_i`` are accepted, and the corresponding
-coefficients are summed together. "Mirrored" indices `(q,r)` and `(r,q)` (where
-`r` and `q` are `VariableIndex`es) are considered duplicates; only one need be
-specified.
+defined as ``\\frac{1}{2}x^\\top Q_i x + a_i^\\top x + b_i``, where:
+
+ * ``\\frac{1}{2}x^\\top Q_i x`` is the symmetric matrix given by the
+   [`VectorQuadraticTerm`](@ref) elements in `quadratic_terms` with
+   `output_index == i`
+ * ``a_i^\\top x`` is the sparse vector given by the [`VectorAffineTerm`](@ref)
+   elements in `affine_terms` with `output_index == i`
+ * ``b_i`` is a scalar given by `constants[i]`
+
+## Duplicates
+
+Duplicate indices in `quadratic_terms` and `affine_terms` with the same
+`output_index` are handled in the same manner as duplicates in
+[`ScalarQuadraticFunction`](@ref).
+
+## Example
+
+```jldoctest
+julia> import MathOptInterface as MOI
+
+julia> x = MOI.VariableIndex(1);
+
+julia> y = MOI.VariableIndex(2);
+
+julia> constants = [4.0, 5.0];
+
+julia> affine_terms = [
+           MOI.VectorAffineTerm(Int64(1), MOI.ScalarAffineTerm(2.0, x)),
+           MOI.VectorAffineTerm(Int64(2), MOI.ScalarAffineTerm(3.0, x)),
+       ];
+
+julia> quad_terms = [
+        MOI.VectorQuadraticTerm(Int64(1), MOI.ScalarQuadraticTerm(2.0, x, x)),
+        MOI.VectorQuadraticTerm(Int64(2), MOI.ScalarQuadraticTerm(3.0, x, y)),
+           ];
+
+julia> f = MOI.VectorQuadraticFunction(quad_terms, affine_terms, constants)
+MathOptInterface.VectorQuadraticFunction{Float64}(MathOptInterface.VectorQuadraticTerm{Float64}[MathOptInterface.VectorQuadraticTerm{Float64}(1, MathOptInterface.ScalarQuadraticTerm{Float64}(2.0, MathOptInterface.VariableIndex(1), MathOptInterface.VariableIndex(1))), MathOptInterface.VectorQuadraticTerm{Float64}(2, MathOptInterface.ScalarQuadraticTerm{Float64}(3.0, MathOptInterface.VariableIndex(1), MathOptInterface.VariableIndex(2)))], MathOptInterface.VectorAffineTerm{Float64}[MathOptInterface.VectorAffineTerm{Float64}(1, MathOptInterface.ScalarAffineTerm{Float64}(2.0, MathOptInterface.VariableIndex(1))), MathOptInterface.VectorAffineTerm{Float64}(2, MathOptInterface.ScalarAffineTerm{Float64}(3.0, MathOptInterface.VariableIndex(1)))], [4.0, 5.0])
```

Comment:
We should really improve the printing of functions, debugging it awful atm

Reply:
Yes. But that can be a separate PR.

I even wonder about something like:
```Julia
ScalarAffineFunction:
    constant: 1.0
    terms:
      VariableIndex(1) => 2.0
      VariableIndex(2) => 2.0
      VariableIndex(3) => 2.0
      VariableIndex(4) => 2.0
```
Rather than trying to pretty-print it all.

Reply:
I would prefer pretty printing for the oneline version as it is sometimes coefficients of a multivariate polynomials that is trying to print itself in oneline.

Reply:
For the long version, your suggestion seems nice

Reply:
Open an issue? What is the "long version"?